components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/NestedWebView.kt

```diff
@@ -45,6 +47,19 @@ class NestedWebView(context: Context) : WebView(context), NestedScrollingChild {
     @VisibleForTesting
     internal var childHelper: NestedScrollingChildHelper = NestedScrollingChildHelper(this)
 
+    /**
+     * @return Integer indicating how user's MotionEvent was handled.
+     *
+     * There must be a 1-1 relation between this values and [EngineView.InputResult]'s.
+     */
+    internal val inputResult: Int
```

Comment:
I think you can move `inputResult` and `eventHandler` to the `SystemEngineView` since that's where the `EngineView.getInputResult` is implemented and needed.

Then you can remove the type check as well:
```kotlin
(session?.webView as? NestedWebView)?.let { nestedWebView ->
  // ...
}
```

Reply:
The value for `eventHandled` is only calculated and set in `NestedWebView`. Don't see how I can extract it from there to `SystemEngineView`.
And if I am to keep `eventHandled` in `NestedWebView` I'll still need the safe cast because `session.webView` returns `WebView` 
..which is indeed the parent of `NestedWebView` but not really sure if I should change the declared type of `session.webView`.

Reply:
Yes, you're right! Ignore me. 😄

Reply:
I thought the NestedWebView has a reference to the SystemEngineView, but it was the other way around.

Reply:
NP, thanks!